### ISSUE TITLE
Hello, I have loaded the DeDRM tools and I am still getting an error loading or converting

### ISSUE THREAD

**Issue by tonycoorey** [apprenticeharper/DeDRM_tools#3]
I have run this in debug as suggested and here is the debug info. Thanks in advance for your help.

C:\Users\tony.coorey\Documents>dir
 Volume in drive C is Default
 Volume Serial Number is 888C-0193

 Directory of C:\Users\tony.coorey\Documents

Tue 24 Mar 15  15:08    <DIR>          .
Tue 24 Mar 15  15:08    <DIR>          ..
Tue 24 Mar 15  15:08    <DIR>          Calibre Library
Tue 11 Nov 14  09:03    <DIR>          CRM
Tue 24 Mar 15  05:09         1,621,311 Mightier Than The Sword The Clifton Chron
icles 5.epub
Thu 03 Jul 14  10:23         6,479,545 Windows 8 Guide.docx
               2 File(s)      8,100,856 bytes
               4 Dir(s)  163,353,190,400 bytes free

C:\Users\tony.coorey\Documents>cd calibre library

C:\Users\tony.coorey\Documents\Calibre Library>dir
 Volume in drive C is Default
 Volume Serial Number is 888C-0193

 Directory of C:\Users\tony.coorey\Documents\Calibre Library

Tue 24 Mar 15  15:08    <DIR>          .
Tue 24 Mar 15  15:08    <DIR>          ..
Tue 24 Mar 15  13:48    <DIR>          DeDRM
Tue 24 Mar 15  09:23    <DIR>          John Schember
Tue 24 Mar 15  15:08           133,120 metadata.db
Tue 24 Mar 15  15:08            12,995 metadata_db_prefs_backup.json
Tue 24 Mar 15  05:09         1,621,311 Mightier Than The Sword The Clifton Chron
icles 5.epub
               3 File(s)      1,767,426 bytes
               4 Dir(s)  163,353,190,400 bytes free

C:\Users\tony.coorey\Documents\Calibre Library>c:\program files\calibre2\calibre
-debug -g
'c:\program' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\tony.coorey\Documents\Calibre Library>cd\program files

C:\Program Files>dir
 Volume in drive C is Default
 Volume Serial Number is 888C-0193

 Directory of C:\Program Files

Tue 24 Mar 15  08:29    <DIR>          .
Tue 24 Mar 15  08:29    <DIR>          ..
Tue 24 Mar 15  08:30    <DIR>          Calibre2
Thu 13 Feb 14  09:40    <DIR>          Common Files
Thu 06 Feb 14  11:04    <DIR>          DellTPad
Thu 06 Feb 14  11:03    <DIR>          DIFX
Wed 18 Dec 13  16:44    <DIR>          DVD Maker
Thu 13 Mar 14  13:47    <DIR>          Google
Thu 06 Feb 14  11:04    <DIR>          IDT
Thu 26 Feb 15  15:01    <DIR>          Internet Explorer
Thu 13 Feb 14  09:38    <DIR>          Microsoft Analysis Services
Fri 13 Mar 15  10:26    <DIR>          Microsoft Mouse and Keyboard Center
Thu 13 Feb 14  09:40    <DIR>          Microsoft Office
Thu 24 Jul 14  17:35    <DIR>          Microsoft Silverlight
Wed 18 Dec 13  12:29    <DIR>          Microsoft SQL Server
Tue 14 Jul 09  16:32    <DIR>          MSBuild
Thu 06 Feb 14  11:04    <DIR>          NVIDIA Corporation
Tue 14 Jul 09  16:32    <DIR>          Reference Assemblies
Thu 06 Feb 14  10:50    <DIR>          STMicroelectronics
Mon 10 Feb 14  11:48    <DIR>          uvnc bvba
Thu 19 Dec 13  09:55    <DIR>          Windows Defender
Thu 26 Feb 15  15:01    <DIR>          Windows Journal
Wed 18 Dec 13  16:44    <DIR>          Windows Mail
Mon 02 Mar 15  07:34    <DIR>          Windows Media Player
Tue 14 Jul 09  16:32    <DIR>          Windows NT
Wed 18 Dec 13  16:44    <DIR>          Windows Photo Viewer
Wed 18 Dec 13  16:44    <DIR>          Windows Portable Devices
Wed 18 Dec 13  16:44    <DIR>          Windows Sidebar
               0 File(s)              0 bytes
              28 Dir(s)  163,353,145,344 bytes free

C:\Program Files>dir cal*
 Volume in drive C is Default
 Volume Serial Number is 888C-0193

 Directory of C:\Program Files

Tue 24 Mar 15  08:30    <DIR>          Calibre2
               0 File(s)              0 bytes
               1 Dir(s)  163,353,145,344 bytes free

C:\Program Files>dir/w
 Volume in drive C is Default
 Volume Serial Number is 888C-0193

 Directory of C:\Program Files

[.]                                   [..]
[Calibre2]                            [Common Files]
[DellTPad]                            [DIFX]
[DVD Maker]                           [Google]
[IDT]                                 [Internet Explorer]
[Microsoft Analysis Services]         [Microsoft Mouse and Keyboard Center]
[Microsoft Office]                    [Microsoft Silverlight]
[Microsoft SQL Server]                [MSBuild]
[NVIDIA Corporation]                  [Reference Assemblies]
[STMicroelectronics]                  [uvnc bvba]
[Windows Defender]                    [Windows Journal]
[Windows Mail]                        [Windows Media Player]
[Windows NT]                          [Windows Photo Viewer]
[Windows Portable Devices]            [Windows Sidebar]
               0 File(s)              0 bytes
              28 Dir(s)  163,353,145,344 bytes free

C:\Program Files>cd calibre2

C:\Program Files\Calibre2>dir/w
 Volume in drive C is Default
 Volume Serial Number is 888C-0193

 Directory of C:\Program Files\Calibre2

[.]                        [..]
calibre-complete.exe       calibre-customize.exe
calibre-debug.exe          calibre-eject.exe
calibre-launcher.dll       calibre-parallel.exe
calibre-server.exe         calibre-smtp.exe
calibre.exe                calibredb.exe
[DLLs]                     ebook-convert.exe
ebook-device.exe           ebook-edit.exe
ebook-meta.exe             ebook-polish.exe
ebook-viewer.exe           fetch-ebook-metadata.exe
LICENSE                    lrf2lrs.exe
lrfviewer.exe              lrs2lrf.exe
markdown-calibre.exe       [Microsoft.VC90.CRT]
pdfinfo.exe                pdftohtml.exe
pdftoppm.exe               [plugins2]
pylib.zip                  [qt_plugins]
[resources]                web2disk.exe
              27 File(s)     62,350,976 bytes
               7 Dir(s)  163,353,145,344 bytes free

C:\Program Files\Calibre2>calibre-debug -g
calibre 2.22 [64bit]  isfrozen: True is64bit: True
Windows-7-6.1.7601-SP1 Windows ('64bit', 'WindowsPE')
('Windows', '7', '6.1.7601')
Python 2.7.9
Windows: ('7', '6.1.7601', 'SP1', 'Multiprocessor Free')
Successfully initialized third party plugins: DeDRM
Starting up...
Started up in 2.34 seconds with 1 books
DeDRM v6.2.0: Trying to decrypt Mightier Than The Sword The Clifton Chronicles 5
.epub
DeDRM v6.2.0: Verifying zip archive integrity
DeDRM v6.2.0: Mightier Than The Sword The Clifton Chronicles 5.epub is a secure
Adobe Adept ePub
DeDRM v6.2.0: Looking for new default Adobe Digital Editions Keys after 0.7 seco
nds
DeDRM v6.2.0: Exception when getting default Adobe Key after 0.7 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 310, in ePubDecrypt
  File "calibre_plugins.dedrm.adobekey", line 377, in adeptkeys
ADEPTError: Adobe Digital Editions not activated
DeDRM v6.2.0: Ultimately failed to decrypt after 0.7 seconds.
Read the FAQs at Alf's blog: http://apprenticealf.wordpress.com/
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 155, in _run_filetype_plugi
ns
  File "calibre_plugins.dedrm.__init__", line 587, in run
  File "calibre_plugins.dedrm.__init__", line 364, in ePubDecrypt
DeDRMError: DeDRM v6.2.0: Ultimately failed to decrypt after 0.7 seconds.
Added Mightier Than The Sword: The Clifton Chronicles 5 to db in: 2.1
Added 1 books in 4.7 seconds
Worker Launch took: 0.338000059128
Job: 1 Convert book 1 of 1 (Mightier Than The Sword: The Clifton Chronicles 5) f
inished
Convert book 1 of 1 (Mightier Than The Sword: The Clifton Chronicles 5)
        Resolved conversion options
        calibre version: 2.22.0
        {'asciiize': False,
         'author_sort': None,
         'authors': None,
         'base_font_size': 0.0,
         'book_producer': None,
         'change_justification': u'original',
         'chapter': u"//*[((name()='h1' or name()='h2') and re:test(., '\\s*((ch
apter|book|section|part)\\s+)|((prolog|prologue|epilogue)(\\s+|$))', 'i')) or @c
lass = 'chapter']",
         'chapter_mark': u'pagebreak',
         'comments': None,
         'cover': None,
         'debug_pipeline': None,
         'dehyphenate': True,
         'delete_blank_paragraphs': True,
         'disable_font_rescaling': False,
         'dont_compress': False,
         'duplicate_links_in_toc': False,
         'embed_all_fonts': False,
         'embed_font_family': None,
         'enable_heuristics': False,
         'expand_css': False,
         'extra_css': None,
         'extract_to': None,
         'filter_css': u'',
         'fix_indents': True,
         'font_size_mapping': None,
         'format_scene_breaks': True,
         'html_unwrap_factor': 0.4,
         'input_encoding': None,
         'input_profile': <calibre.customize.profiles.InputProfile object at 0x0
0000000025247B8>,
         'insert_blank_line': False,
         'insert_blank_line_size': 0.5,
         'insert_metadata': False,
         'isbn': None,
         'italicize_common_cases': True,
         'keep_ligatures': False,
         'language': None,
         'level1_toc': None,
         'level2_toc': None,
         'level3_toc': None,
         'line_height': 0.0,
         'linearize_tables': False,
         'margin_bottom': 5.0,
         'margin_left': 5.0,
         'margin_right': 5.0,
         'margin_top': 5.0,
         'markup_chapter_headings': True,
         'max_toc_links': 50,
         'minimum_line_height': 120.0,
         'mobi_file_type': u'old',
         'mobi_ignore_margins': False,
         'mobi_keep_original_images': False,
         'mobi_toc_at_start': False,
         'no_chapters_in_toc': False,
         'no_inline_navbars': True,
         'no_inline_toc': False,
         'output_profile': <calibre.customize.profiles.KindlePaperWhiteOutput ob
ject at 0x0000000002524E10>,
         'page_breaks_before': u'/',
         'personal_doc': u'[PDOC]',
         'prefer_author_sort': False,
         'prefer_metadata_cover': False,
         'pretty_print': False,
         'pubdate': None,
         'publisher': None,
         'rating': None,
         'read_metadata_from_opf': u'C:\\Users\\TONY~1.COO\\AppData\\Local\\Temp
\\calibre_vbb7rp\\bikdmi.opf',
         'remove_fake_margins': True,
         'remove_first_image': False,
         'remove_paragraph_spacing': False,
         'remove_paragraph_spacing_indent_size': 1.5,
         'renumber_headings': True,
         'replace_scene_breaks': u'',
         'search_replace': '[]',
         'series': None,
         'series_index': None,
         'share_not_sync': False,
         'smarten_punctuation': False,
         'sr1_replace': None,
         'sr1_search': None,
         'sr2_replace': None,
         'sr2_search': None,
         'sr3_replace': None,
         'sr3_search': None,
         'start_reading_at': None,
         'subset_embedded_fonts': False,
         'tags': None,
         'timestamp': None,
         'title': None,
         'title_sort': None,
         'toc_filter': None,
         'toc_threshold': 6,
         'toc_title': None,
         'unsmarten_punctuation': False,
         'unwrap_lines': True,
         'use_auto_toc': False,
         'verbose': 2}
        InputFormatPlugin: EPUB Input running
        on C:\Users\TONY~1.COO\AppData\Local\Temp\calibre_vbb7rp\kcdmym.epub
        Python function terminated unexpectedly
          kcdmym.epub (Error Code: 1)
        Traceback (most recent call last):
          File "site.py", line 132, in main
          File "site.py", line 109, in run_entry_point
          File "site-packages\calibre\utils\ipc\worker.py", line 193, in main
          File "site-packages\calibre\gui2\convert\gui_conversion.py", line 31,
in gui_convert_override
          File "site-packages\calibre\gui2\convert\gui_conversion.py", line 25,
in gui_convert
          File "site-packages\calibre\ebooks\conversion\plumber.py", line 1041,
in run
          File "site-packages\calibre\customize\conversion.py", line 241, in __c
all__
          File "site-packages\calibre\ebooks\conversion\plugins\epub_input.py",
line 218, in convert
        calibre.ebooks.DRMError: kcdmym.epub

C:\Program Files\Calibre2>^A

**Comment by apprenticeharper** [comment]
"ADEPTError: Adobe Digital Editions not activated"

It seems that you have not registered (activated) your copy of Adobe Digital Editions with your Adpbe ID before using it to download DRMed ebooks.

So the tools can't find a decryption key, and your book(s) are tied to this particular installation of Adobe Digital Editions.

You might get more help at Apprentice Alf's blog, but I don't remember ever seeing a solution for this.

Closed. Not a bug.

**Comment by Andreacroci** [comment]
Hi, sorry for posting in a long closed thread, but I haven't seen this been carried forward anywhere else. Feel free to move this comment if necessary.

Having said that, I have also had this message (Ubuntu 16.04, Wine 3.0, ADE 1.7, the default installed via Winetricks) and it's indeed not a bug, but I have been able to understand it a little more: because  I haven't been able to generate a key from the configuration window of the plugin within Calibre, no matter which WINEPREFIX I used, I tried running the script from Console instead.

The above message is due to the fact that Wine makes a .wine folder in your home directory when it gets installed and all the Pythons get installed there. Then Winetricks makes a WINEPREFIX under .local/share/wineprefixes/adobe_diged where it installs ADE.

What happens is, if the adobekey.py script is run with the ADE-Wineprefix, it can't find python and it complains. If it's run with the .wine Wineprefix, then it can't find ADE and the message "ADEPTError: Adobe Digital Editions not activated" is thrown, although ADE is indeed installed and activated.

So basically, if you want to run the script from Console, you have to choose the WINEPREFIX where ADE is installed, so that it finds the key, and then tell wine where to find python by giving the full path to it (~.wine/drive_c/windows/... depending on the installation).
